osu.Game.Rulesets.Taiko/Mods/TaikoModHidden.cs

```diff
@@ -40,15 +40,26 @@ public void ApplyToDrawableRuleset(DrawableRuleset<TaikoHitObject> drawableRules
 
         protected override void ApplyIncreasedVisibilityState(DrawableHitObject hitObject, ArmedState state)
         {
-            ApplyNormalVisibilityState(hitObject, state);
+            applyHiddenState(hitObject, state, true);
         }
 
         protected override void ApplyNormalVisibilityState(DrawableHitObject hitObject, ArmedState state)
+        {
+            applyHiddenState(hitObject, state, false);
+        }
+
+        private void applyHiddenState(DrawableHitObject hitObject, ArmedState state, bool increaseVisibility)
         {
             switch (hitObject)
             {
                 case DrawableDrumRollTick:
                 case DrawableHit:
+                    if (increaseVisibility)
+                    {
+                        hitObject.FadeIn(hitObject.HitObject.HitWindows.WindowFor(HitResult.Miss));
```

Comment:
What is the reason for using the miss window as the fade-in duration here? I don't see any. Why is there any fade-in at all even?

Reply:
Neither makes much sense to me. Why fade in at all?